src/templates/template-common/utils.py

```diff
@@ -156,7 +171,7 @@ def setup_output_dir(config: Any, rank: int) -> Path:
         path.mkdir(parents=True, exist_ok=True)
         config.output_dir = path.as_posix()
 
-    return Path(idist.broadcast(config.output_dir, src=0))
+    return Path(idist.broadcast(str(config.output_dir), src=0))
```

Comment:
Also it seems that we can have problems changing the datatype once specified in the config for `Hydra Compose API` so I added `__str__` representation of the path here.

Reply:
I tried removing `str` in colab and do not see any issues with that. I still do not understand why we need to modify the code here.

> we can have problems changing the datatype

Which problems exactly you have ?

Reply:
Some tests seem to fail like here https://github.com/pytorch-ignite/code-generator/actions/runs/5963864652/job/16177915022#step:17:64